File: HW4/Extras/graphme.py
#!/usr/bin/env python

import matplotlib.pyplot as plt
import math
import sys

shape = 'sine'

def main(arg):

    pointarray = []
    timearray = []

    print arg

    if arg == 'triangle':
        wave_period = 1./5    #s/wave
        duration = wave_period
    elif arg == 'sine':
        wave_period = 1./10
        duration = wave_period
    else:
        print "arg invalid"
        sys.exit()

    samp = 1000.  #samp/s
    numsamps = samp*duration
    time = 0.


    # for i in range(1, int(numsamps)):
    #     if shape == 'triangle':
    #         pointarray.append(int(1024./3.3 * triangle(3.3, time, wave_period)))
    #     elif shape == 'sine':
    #         pointarray.append(int(1024./3.3 * sine(1.65, time, wave_period)))
    #     timearray.append(time)
    #     time += 1./samp

    for i in range(0, int(numsamps)):
        if arg == 'triangle':
            pointarray.append(triangle(3.3, time, wave_period))
        elif arg == 'sine':
            pointarray.append(sine(1.65, time, wave_period))
        timearray.append(time)
        time += 1./samp


    plt.plot(pointarray)
    plt.show()


def triangle(amp, time, wave_period):
    '''
    Given the desired amplitude, time, and waveform wave_period, returns the voltage of the reference wave with that amplitude and wave_period at the given time.
    '''

    return (amp*2)/wave_period * (wave_period/2 - math.fabs(time % (wave_period) - wave_period/2))



def sine(amp, time, wave_period):
    '''
    Given the desired amplitude, time, and waveform wave_period, returns the voltage of the reference wave with that amplitude and wave_period at the given time.
    '''

    phase = 0.
    return amp * math.sin(2.*math.pi*1./wave_period*time+phase) + 1.65








if __name__ == '__main__':
    if len(sys.argv) > 1:
        main(sys.argv[1])
    else:
        print "Argument: 'triangle or sine'"

File: HW4/HW4.X/graphme.c
#include<math.h>
#include<stdio.h>




#define shape 's'




double triangle_gen(double amplitude, double time, double wave_period);
double sin_gen(double amplitude, double time, double wave_period);



int main()
{

    double wave_period = 0;             //Seconds per wave
    double duration = 0;                //Duration of the trajectory
    double samp = 1000.;                //Samples per second
    double time = 0;                    //Time stamp


    if (shape == 't')
    {
        wave_period = 1.0/5.0;
        duration = wave_period;
    }
    else if (shape == 's')
    {
        wave_period = 1.0/10.0;
        duration = wave_period;
    }
    else
    {
        printf("arg invalid");
        return 1;
    }

    double numsamps = samp * duration;  //Total Samples
    double pointarray[(int)numsamps];
    double timearray[(int)numsamps];

    for(int i=0; i<(int)numsamps; i++)
    {
        if (shape == 't')
            pointarray[i] = triangle_gen(3.3, time, wave_period);
        else if (shape == 's')
            pointarray[i] = sin_gen(1.65, time, wave_period);
        timearray[i] = time;
        time = time + 1.0/samp;          //Seconds per sample
    }

    for(int i=0; i<(int)numsamps; i++)
        printf("%f \n", pointarray[i]);

    return 0;
}

double triangle_gen(double amplitude, double time, double wave_period)
{
    return (amplitude * 2/wave_period * (wave_period/2 - fabs(fmod(time, wave_period) - wave_period/2)));
}



double sin_gen(double amplitude, double time, double wave_period)
{
    float phase = 0.0;
    return amplitude * sin(2.0 * 3.14159 * 1.0/wave_period * time + phase) + 1.65;
}
